pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -189,3 +203,21 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
 
 	return err
 }
+
+// createBootstrapFirewallRules creates the rules needed for the bootstrap node.
+func createBootstrapFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, network string, api string) error {
+	projectID := in.InstallConfig.Config.Platform.GCP.ProjectID
+	firewallName := fmt.Sprintf("%s-bootstrap-in-ssh", in.InfraID)
+	srcTags := []string{}
+	bootstrapTag := fmt.Sprintf("%s-control-plane", in.InfraID)
+	targetTags := []string{bootstrapTag}
+	var srcRanges []string
+	machineCIDR := in.InstallConfig.Config.Networking.MachineNetwork[0].CIDR.String()
+	if api != "" {
```

Comment:
I think this condition should be based on the publishing strategy. You're using `gcpCluster.Spec.ControlPlaneEndpoint.Host` instead, but I would think in a private cluster `Host` would have the ip address of the internal LB... not be empty. Of course that is conjecture because you're in process of writing internal capg :D 

But I think setting src range explicitly on publishing strategy is the way to go

Reply:
Changed to use publishing strategy.